File: leetcode/Arrays/leetcode-167/167.go
package arrays

func twoSum(numbers []int, target int) []int {
	if len(numbers) < 2 {
		return nil
	}

	left, right := 0, len(numbers)-1
	for left < right {
		if numbers[left]+numbers[right] < target {
			left++
		} else if numbers[left]+numbers[right] > target {
			right--
		} else {
			break
		}
	}

	if numbers[left]+numbers[right] == target && left < right {
		return []int{left + 1, right + 1}
	}
	return nil
}

File: leetcode/Arrays/leetcode-167/167.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        if (numbers == null || numbers.length < 2) {
            return new int[0];
        }

        int left = 0, right = numbers.length - 1;
        while (left < right) {
            if (numbers[left] + numbers[right] < target) {
                left++;
            } else if (numbers[left] + numbers[right] > target) {
                right--;
            } else {
                break;
            }
        }

        if (numbers[left] + numbers[right] == target && left < right) {
            return new int[]{left + 1, right + 1};
        }
        return new int[0];
    }
}